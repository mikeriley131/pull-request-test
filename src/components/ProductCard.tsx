export default function ProductCard({ productData }) {
  return (
    <div>
      <ul>
        <li>{productData.imgSrc}</li>
        <li>{productData.name}</li>
        <li>{productData.author}</li>
        <li>{productData.description}</li>
        <li>add to cart button</li>
        <li>like button</li>
        <li>- full or empty heart emoji</li>
        <li>product features list</li>
      </ul>
    </div>
  );
}
