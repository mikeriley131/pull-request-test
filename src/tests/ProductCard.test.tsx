import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import ProductCard from '../components/ProductCard';

describe('ProductCard', () => {
  it('renders', () => {
    render(
      <ProductCard
        productData={{
          name: 'Test Product',
          description: 'This is a test product',
          author: 'Some Author',
          imgSrc: '/test.png',
          features: ['one', 'two', 'three'],
        }}
      />,
    );

    expect(screen.getByText('Test Product')).toBeTruthy();

    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it('likes product when clicked', () => {
    render(
      <ProductCard
        productData={{
          name: 'Another Product',
          description: 'Desc',
          imgSrc: '/test2.png',
          features: ['x'],
          author: 'Some Author',
        }}
      />,
    );

    fireEvent.click(screen.getByText('Another Product'));
    expect(screen.getByText('❤️')).toBeTruthy();
  });

  it('adds to cart', async () => {
    render(
      <ProductCard
        productData={{
          name: 'Cart Product',
          author: 'Some Author',
          description: 'Cart Desc',
          imgSrc: '/cart.png',
          features: [],
        }}
      />,
    );

    fireEvent.click(screen.getByText('Add'));
  });

  it('renders three features for each card', () => {
    const product = {
      name: 'Feature Product',
      author: 'Some Author',
      description: 'Desc',
      imgSrc: '/feature.png',
      features: ['feature1', 'feature2', 'feature3'],
    };

    render(<ProductCard productData={product} />);

    expect(screen.getByText('feature2')).toBeTruthy();
  });
});
