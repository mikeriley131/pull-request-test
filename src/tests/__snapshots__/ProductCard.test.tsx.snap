// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard > renders 1`] = `"<div><div class="product-card"><img src="/test.png"><h2>Test Product</h2><h5>Some Author</h5><p>This is a test product</p><ul class="aligned-list"><li>one</li><li>two</li><li>three</li></ul><button>Add</button><div>â™¡</div></div></div>"`;
